core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -427,12 +439,11 @@ private static void completeSatisfiedFutures(RequestLogFuture[] satisfiedFutures
     }
 
     @Nullable
-    private RequestLogFuture[] removeSatisfiedFutures() {
+    private RequestLogFuture[] removeSatisfiedFutures(List<RequestLogFuture> pendingFutures) {
```

Comment:
So that we can make sure that the `pendingFutures` referred to in this method is the one in the parameter

```suggestion
    private static RequestLogFuture[] removeSatisfiedFutures(List<RequestLogFuture> pendingFutures) {
```

Reply:
When I change the method to static, I get an error because the flag variable in this method is a local variable.
If you are confused with a global variable name, how about using a different name?
instead of making the method static e.g. using localPendingFutures in the method parameter
p.s.
I changed the flag to static beforehand and it gives me a bunch of errors.😭

Reply:
Gentle ping

Reply:
Sorry about the delay 😅   Pushed a small commit because I felt sorry about not responding earlier 🙏   Let me know if it doesn't make sense 🙏

Reply:
LGTM👍